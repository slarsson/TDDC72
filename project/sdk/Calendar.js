import React, { useState, useRef } from 'react';
import Icon from 'react-native-vector-icons/FontAwesome';
import {
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
} from 'react-native';

const gridStyles = StyleSheet.create({
  current: {
    backgroundColor: 'red'
  },
  currentText: {
    color: 'green'
  },
  today: {
    backgroundColor: 'orange'
  },
  todayText: {
    color: 'red'
  },
  other: {
    backgroundColor: 'blue'
  },
  otherText: {
    color: 'pink'
  }
});

const test = (year, month, today, cb) => {
  let lastMonth = new Date(year, month, 0).getDate();
  let currentMonth = new Date(year, (month + 1), 0).getDate();
  let offset = new Date(year, month, 1).getDay();
  
  let rows = [];
  let day, type;

  if (offset > 0) {
    day = lastMonth - (offset - 1);
    type = true;
  } else {
    day = 1;
    type = false;
  }
  
  let styleBox, styleText;
  let count = 0; 

  for (let j = 0; j <= 5; j++) {
    let row = [];
    for (let i = 1; i <= 7; i++) {
      if (type) {
        styleBox = gridStyles.other;
        styleText = gridStyles.otherText;
      } else if (day == today) {
        styleBox = gridStyles.today;
        styleText = gridStyles.todayText;
      } else {
        styleBox = gridStyles.current;
        styleText = gridStyles.currenText;
      }
      
      let x = {
        day: day,
        month: type ? (day > 15 ? -1 : 1) : 0,
      };

      row.push(
        <TouchableOpacity
          key={Math.random() + Math.random()}
          style={[styles.day, styleBox]}
          onPress={() => cb(x)}
        >
          <Text style={styleText}>{day}</Text>
        </TouchableOpacity>
      );
      
      day++;
      count++;
      if (type) {
        if (day > lastMonth) {
          type = false;
          day = 1;
        }
      } else {
        if (day > currentMonth) {
          day = 1;
          type = true;     
        }
      }
    }
    rows.push(<View key={j} style={{flexDirection: 'row'}}>{row}</View>)
  }
  return rows;
}

const MONTHS = ['January','February','March','April','May','June','July','August','September','October','November','December'];
const DAYS = ['SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT'];

const Calender = ({currentYear, currentMonth, currentDay}) => {
  const [year, setYear] = useState(currentYear);
  const [month, setMonth] = useState(currentMonth);
  
  const today = useRef({
    day: currentDay,
    month: currentMonth,
    year: currentYear
  });

  const next = () => {
    if (month == 11) {
      setMonth(0);  
      setYear(year + 1);
    } else {
      setMonth(month + 1);
    }
  };

  const prev = () => {
    if(month == 0) {
      setMonth(11);
      setYear(year - 1);
    } else {
      setMonth(month - 1);
    }
  };

  return (
    <View style={{margin: 20}}>
      <View style={styles.dateHeaderPicker}>
        <TouchableOpacity
          onPress={() => prev()}
        >
          <Icon name="chevron-left" size={40} color={'gold'} />   
        </TouchableOpacity>     
          <Text style={styles.dateHeader}>{MONTHS[month]} {year}</Text>
        <TouchableOpacity
          onPress={() => next()}
        >
          <Icon name="chevron-right" size={40} color={'gold'} />
        </TouchableOpacity>
      </View>

      <View style={styles.dayHeader}>
        {DAYS.map(item => {
          return <Text key={item} style={styles.dayHeaderItem}>{item}</Text>
        })}
      </View>
      
      <View style={styles.daysContainer}>
        {test(year, month, (year == today.current.year && month == today.current.month ? today.current.day : -1), (x) => {
          if (x.month == 1) {
            next();
          } else if (x.month == -1) {
            prev();
          }
          console.log(x.day);
        })}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  dateHeaderPicker: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom: 10,
  },
  dateHeader: {
    fontSize: 20,
    textAlign: 'center',
  },
  daysContainer: {
    width: '100%',
    backgroundColor: 'red',
  },
  dayHeader: {
    width: '100%',
    flexDirection: 'row',
  },
  dayHeaderItem: {
    width: 100/7 + '%',
    textAlign: 'center',
  },
  day: {
    width: 100/7 + '%',
    backgroundColor: '#ededed',
    aspectRatio: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  
});


export default Calender;