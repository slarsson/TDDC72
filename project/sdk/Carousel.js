import React, { Component, useContext, createContext } from 'react';
import {
  View,
  Text,
  ScrollView,
  Image,
  StyleSheet,
  TouchableOpacity
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

const CarouselContext = createContext(null);

class Carousel extends Component {
  constructor(props) {
    super(props);
  }
  
  render() {
    return (
      <View style={this.props.style}>
        {this.props.title && <Text style={{...styles.containerTitle, ...this.props.titleStyle}}>{this.props.title}</Text>}
        <ScrollView horizontal={true} contentContainerStyle={{alignItems: 'flex-start'}}>
          <CarouselContext.Provider value={{
            callback: this.props.onPress,
            width: this.props.itemWidth ? this.props.itemWidth : 150,
            containerStyle: this.props.itemContainerStyle ? this.props.itemContainerStyle : {},  
            textStyle: this.props.itemTextStyle ? this.props.itemTextStyle : {} 
          }}>
            {this.props.children}
          </CarouselContext.Provider>
        </ScrollView>
      </View>
    );
  }
}

const Item = (props) => {
  const ctx = useContext(CarouselContext);

  return (
    <TouchableOpacity style={{...styles.item, width: ctx.width}} activeOpacity={0.6} onPress={() => {
      if (typeof ctx.callback !== 'function') return;
      ctx.callback({...props});
    }}>
      <Image
        style={styles.image}
        source={{uri: props.url}}
      />
      <View style={{...ctx.containerStyle}}>
        {props.rating && <View style={styles.rating}>
          <Icon name="star" size={25} color={'gold'} />
          <Text style={{...styles.ratingText, color: ctx.textStyle.color ? ctx.textStyle.color : '#fff'}}>{props.rating}</Text>
        </View>}
        {props.name && <Text style={{...styles.title, ...ctx.textStyle}} numberOfLines={1}>{props.name}</Text>}
      </View>
    </TouchableOpacity>
  ); 
};

Carousel.Item = Item;

const styles = StyleSheet.create({
  containerTitle: {
    color: '#fff',
    fontSize: 25,
    paddingLeft: 10,
    paddingTop: 10
  },
  item: {
    flexDirection: 'column',
    margin: 10
  },
  image: {
    minWidth: '100%',
    aspectRatio: 2/3
  },
  rating: {
    padding: 10,
    flexDirection: 'row',
    alignItems: 'center'
  },
  ratingText: {
    paddingLeft: 5,
    fontSize: 20
  },
  title: {
    paddingLeft: 10,
    paddingRight: 10,
    paddingBottom: 15,
    fontSize: 20,
    color: 'white',
  }
});

export default Carousel

