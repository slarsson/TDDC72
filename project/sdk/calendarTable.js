import React from 'react';
import {
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
} from 'react-native';

const calendarTable = (year, month, today, selected, selectedColor, todayColor, cb) => {
  let lastMonth = new Date(year, month, 0).getDate();
  let currentMonth = new Date(year, (month + 1), 0).getDate();
  let offset = new Date(year, month, 1).getDay();
  
  let rows = [];
  let day, type;

  if (offset > 0) {
    day = lastMonth - (offset - 1);
    type = true;
  } else {
    day = 1;
    type = false;
  }
  
  let styleBox, styleText;

  for (let j = 0; j <= 5; j++) {
    let row = [];
    for (let i = 1; i <= 7; i++) {
      if (type) {
        styleBox = {};
        styleText = gridStyles.otherText;
      } else if (day == today) {
        if (todayColor != null) {
          styleBox = {...gridStyles.today, backgroundColor: todayColor};
        } else {
          styleBox = gridStyles.today;
        }
        styleText = gridStyles.todayText;
      } else if (day == selected) {
        if (selectedColor != null) {
          styleBox = {...gridStyles.selected, backgroundColor: selectedColor};
        } else {
          styleBox = gridStyles.selected;
        }
        styleText = gridStyles.selectedText;
      } else {
        styleBox = {};
        styleText = gridStyles.currentText;
      }

      const obj = {
        day: day,
        month: type ? (day > 15 ? -1 : 1) : 0,
      };

      row.push(
        <TouchableOpacity
          key={Math.random() + Math.random()}
          style={gridStyles.day}
          onPress={() => cb(obj)}
        >
          <View style={[gridStyles.tableCell, styleBox]}>
            <Text style={styleText}>{day}</Text>
          </View>
        </TouchableOpacity>
      );
      
      day++;

      if (type) {
        if (day > lastMonth) {
          type = false;
          day = 1;
        }
      } else {
        if (day > currentMonth) {
          day = 1;
          type = true;     
        }
      }
    }
    rows.push(<View key={j} style={{flexDirection: 'row'}}>{row}</View>)
  }
  return rows;
}

const gridStyles = StyleSheet.create({
  currentText: {
    fontWeight: 'bold'
  },
  today: {
    backgroundColor: 'orange',
    borderRadius: 100
  },
  todayText: {
    color: 'white'
  },
  otherText: {
    color: '#808080'
  },
  selected: {
    backgroundColor: 'tomato',
    borderRadius: 100,
  },
  selectedText: {
    color: 'white'
  },
  day: {
    width: 100/7 + '%',
    aspectRatio: 1,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 2
  },
  tableCell: {
    width: '100%',
    height: '100%',
    alignItems: 'center',
    justifyContent: 'center'
  }   
});

export {
  calendarTable
}