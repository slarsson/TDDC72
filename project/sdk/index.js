import Calendar from './Calendar';
import Carousel from './Carousel';

export { 
    Calendar,
    Carousel
};