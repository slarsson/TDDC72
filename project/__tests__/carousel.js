/**
 * @jest-environment jsdom
*/

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import { Carousel } from '../sdk';

// Test if <Carousel/> renders without any items
it('empty carousel renders correctly', () => {
  const tree = renderer.create(<Carousel></Carousel>).toJSON();
  expect(tree).toMatchSnapshot();
});

// Test if <Carousel/> renders with items
it('carousel with items renders correctly', () => {
  const tree = renderer.create(
    <Carousel>
      <Carousel.Item id={1} name={"item 1"} rating={10} url="https://m.media-amazon.com/images/M/MV5BOTFmNzY4Y2UtYTIzYy00MmE2LTk0YzUtYmY2Y2ZlNjQ0OGUxXkEyXkFqcGdeQXVyMjUyNDk2ODc@._V1_UY268_CR1,0,182,268_AL_.jpg"/>
      <Carousel.Item id={2} name={"item 2"} rating={9.8} url="https://m.media-amazon.com/images/M/MV5BMTY2NjA4MjgwOF5BMl5BanBnXkFtZTcwMjI2MzYyNw@@._V1_UY268_CR3,0,182,268_AL_.jpg"/>
      <Carousel.Item id={3} name={"item 3"} rating={9.6} url="https://m.media-amazon.com/images/M/MV5BYTIyYTgyZmMtNGNlMy00ZWQzLWI0YjctOTAxOWJmYmJjMWY1XkEyXkFqcGdeQXVyMTQzMjU1NjE@._V1_UY268_CR3,0,182,268_AL_.jpg"/>
      <Carousel.Item id={4} name={"item 4"} rating={0.3} url="https://duifokus.se/wp-content/uploads/2017/02/Sameblod-affisch.jpg"/>
    </Carousel>
  ).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.children[0].children[0].children.length).toBe(4);
});

// Test if <Carousel/> renders without name and rating prop
it('carousel with items (images only) renders correctly', () => {
  const tree = renderer.create(
    <Carousel>
      <Carousel.Item id={1} url="https://m.media-amazon.com/images/M/MV5BOTFmNzY4Y2UtYTIzYy00MmE2LTk0YzUtYmY2Y2ZlNjQ0OGUxXkEyXkFqcGdeQXVyMjUyNDk2ODc@._V1_UY268_CR1,0,182,268_AL_.jpg"/>
      <Carousel.Item id={2} url="https://m.media-amazon.com/images/M/MV5BMTY2NjA4MjgwOF5BMl5BanBnXkFtZTcwMjI2MzYyNw@@._V1_UY268_CR3,0,182,268_AL_.jpg"/>
      <Carousel.Item id={3} url="https://m.media-amazon.com/images/M/MV5BYTIyYTgyZmMtNGNlMy00ZWQzLWI0YjctOTAxOWJmYmJjMWY1XkEyXkFqcGdeQXVyMTQzMjU1NjE@._V1_UY268_CR3,0,182,268_AL_.jpg"/>
      <Carousel.Item id={4} url="https://duifokus.se/wp-content/uploads/2017/02/Sameblod-affisch.jpg"/>
    </Carousel>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

// Test if <Carousel.Item/> props renders
it('carousel item name + rating renders correctly', () => {
  const testTitle = 'my test title';
  const testRating = 123;

  const tree = renderer.create(
    <Carousel>
      <Carousel.Item id={1} name={testTitle} rating={testRating} url="https://m.media-amazon.com/images/M/MV5BOTFmNzY4Y2UtYTIzYy00MmE2LTk0YzUtYmY2Y2ZlNjQ0OGUxXkEyXkFqcGdeQXVyMjUyNDk2ODc@._V1_UY268_CR1,0,182,268_AL_.jpg"/>
    </Carousel>
  ).toJSON();

  const title = tree.children[0].children[0].children[0].children[1].children[1].children[0];
  const rating = tree.children[0].children[0].children[0].children[1].children[0].children[1].children[0];

  expect(title).toBe(testTitle); 
  expect(rating).toBe(testRating.toString());
});

// Test if <Carousel.Item/> width has been set successfully
it('carousel item width set ok', () => {
  const itemWidth = 123;

  const tree = renderer.create(
    <Carousel
      itemWidth={itemWidth}
    >
      <Carousel.Item id={1} name={'test'} rating={0} url="https://m.media-amazon.com/images/M/MV5BOTFmNzY4Y2UtYTIzYy00MmE2LTk0YzUtYmY2Y2ZlNjQ0OGUxXkEyXkFqcGdeQXVyMjUyNDk2ODc@._V1_UY268_CR1,0,182,268_AL_.jpg"/>
    </Carousel>
  ).toJSON();

  const width = tree.children[0].children[0].children[0].props.style.width;
  expect(width).toBe(itemWidth);
});

// Test if <Carousel/> callback works
it('carousel item callback props', () => {
  const testItem1 = {
    id: 1,
    name: 'my name 1',
    rating: 123
  };

  const testItem2 = {
    id: 2,
    name: 'my name 2',
    rating: 0.2
  };

  const mockCallBack = jest.fn();
  let carousel = mount(
    <Carousel
      onPress={mockCallBack}
    >
      <Carousel.Item 
        id={testItem1.id} 
        name={testItem1.name} 
        rating={testItem1.rating} 
        url="https://m.media-amazon.com/images/M/MV5BOTFmNzY4Y2UtYTIzYy00MmE2LTk0YzUtYmY2Y2ZlNjQ0OGUxXkEyXkFqcGdeQXVyMjUyNDk2ODc@._V1_UY268_CR1,0,182,268_AL_.jpg"
      />
      <Carousel.Item 
        id={testItem2.id} 
        name={testItem2.name} 
        rating={testItem2.rating} 
        url="https://m.media-amazon.com/images/M/MV5BOTFmNzY4Y2UtYTIzYy00MmE2LTk0YzUtYmY2Y2ZlNjQ0OGUxXkEyXkFqcGdeQXVyMjUyNDk2ODc@._V1_UY268_CR1,0,182,268_AL_.jpg"
      />
    </Carousel>
  );
  
  // test first child
  carousel.find('Item').at(0).simulate('click');
  expect(mockCallBack.mock.calls[0][0].id).toBe(testItem1.id);
  expect(mockCallBack.mock.calls[0][0].name).toBe(testItem1.name);
  expect(mockCallBack.mock.calls[0][0].rating).toBe(testItem1.rating);
  
  // test second child
  carousel.find('Item').at(1).simulate('click');
  expect(mockCallBack.mock.calls[1][0].id).toBe(testItem2.id);
  expect(mockCallBack.mock.calls[1][0].name).toBe(testItem2.name);
  expect(mockCallBack.mock.calls[1][0].rating).toBe(testItem2.rating);
});
