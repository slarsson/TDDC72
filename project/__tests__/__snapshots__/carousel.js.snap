// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`carousel with items (images only) renders correctly 1`] = `
<View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "flex-start",
      }
    }
    horizontal={true}
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "column",
            "margin": 10,
            "opacity": 1,
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://m.media-amazon.com/images/M/MV5BOTFmNzY4Y2UtYTIzYy00MmE2LTk0YzUtYmY2Y2ZlNjQ0OGUxXkEyXkFqcGdeQXVyMjUyNDk2ODc@._V1_UY268_CR1,0,182,268_AL_.jpg",
            }
          }
          style={
            Object {
              "aspectRatio": 0.6666666666666666,
              "minWidth": "100%",
            }
          }
        />
        <View
          style={Object {}}
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "column",
            "margin": 10,
            "opacity": 1,
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://m.media-amazon.com/images/M/MV5BMTY2NjA4MjgwOF5BMl5BanBnXkFtZTcwMjI2MzYyNw@@._V1_UY268_CR3,0,182,268_AL_.jpg",
            }
          }
          style={
            Object {
              "aspectRatio": 0.6666666666666666,
              "minWidth": "100%",
            }
          }
        />
        <View
          style={Object {}}
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "column",
            "margin": 10,
            "opacity": 1,
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://m.media-amazon.com/images/M/MV5BYTIyYTgyZmMtNGNlMy00ZWQzLWI0YjctOTAxOWJmYmJjMWY1XkEyXkFqcGdeQXVyMTQzMjU1NjE@._V1_UY268_CR3,0,182,268_AL_.jpg",
            }
          }
          style={
            Object {
              "aspectRatio": 0.6666666666666666,
              "minWidth": "100%",
            }
          }
        />
        <View
          style={Object {}}
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "column",
            "margin": 10,
            "opacity": 1,
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://duifokus.se/wp-content/uploads/2017/02/Sameblod-affisch.jpg",
            }
          }
          style={
            Object {
              "aspectRatio": 0.6666666666666666,
              "minWidth": "100%",
            }
          }
        />
        <View
          style={Object {}}
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`carousel with items renders correctly 1`] = `
<View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "flex-start",
      }
    }
    horizontal={true}
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "column",
            "margin": 10,
            "opacity": 1,
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://m.media-amazon.com/images/M/MV5BOTFmNzY4Y2UtYTIzYy00MmE2LTk0YzUtYmY2Y2ZlNjQ0OGUxXkEyXkFqcGdeQXVyMjUyNDk2ODc@._V1_UY268_CR1,0,182,268_AL_.jpg",
            }
          }
          style={
            Object {
              "aspectRatio": 0.6666666666666666,
              "minWidth": "100%",
            }
          }
        />
        <View
          style={Object {}}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "gold",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 20,
                  "paddingLeft": 5,
                }
              }
            >
              10
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "white",
                "fontSize": 20,
                "paddingBottom": 15,
                "paddingLeft": 10,
                "paddingRight": 10,
              }
            }
          >
            item 1
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "column",
            "margin": 10,
            "opacity": 1,
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://m.media-amazon.com/images/M/MV5BMTY2NjA4MjgwOF5BMl5BanBnXkFtZTcwMjI2MzYyNw@@._V1_UY268_CR3,0,182,268_AL_.jpg",
            }
          }
          style={
            Object {
              "aspectRatio": 0.6666666666666666,
              "minWidth": "100%",
            }
          }
        />
        <View
          style={Object {}}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "gold",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 20,
                  "paddingLeft": 5,
                }
              }
            >
              9.8
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "white",
                "fontSize": 20,
                "paddingBottom": 15,
                "paddingLeft": 10,
                "paddingRight": 10,
              }
            }
          >
            item 2
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "column",
            "margin": 10,
            "opacity": 1,
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://m.media-amazon.com/images/M/MV5BYTIyYTgyZmMtNGNlMy00ZWQzLWI0YjctOTAxOWJmYmJjMWY1XkEyXkFqcGdeQXVyMTQzMjU1NjE@._V1_UY268_CR3,0,182,268_AL_.jpg",
            }
          }
          style={
            Object {
              "aspectRatio": 0.6666666666666666,
              "minWidth": "100%",
            }
          }
        />
        <View
          style={Object {}}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "gold",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 20,
                  "paddingLeft": 5,
                }
              }
            >
              9.6
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "white",
                "fontSize": 20,
                "paddingBottom": 15,
                "paddingLeft": 10,
                "paddingRight": 10,
              }
            }
          >
            item 3
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexDirection": "column",
            "margin": 10,
            "opacity": 1,
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://duifokus.se/wp-content/uploads/2017/02/Sameblod-affisch.jpg",
            }
          }
          style={
            Object {
              "aspectRatio": 0.6666666666666666,
              "minWidth": "100%",
            }
          }
        />
        <View
          style={Object {}}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "gold",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 20,
                  "paddingLeft": 5,
                }
              }
            >
              0.3
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "white",
                "fontSize": 20,
                "paddingBottom": 15,
                "paddingLeft": 10,
                "paddingRight": 10,
              }
            }
          >
            item 4
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`empty carousel renders correctly 1`] = `
<View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "flex-start",
      }
    }
    horizontal={true}
  >
    <View />
  </RCTScrollView>
</View>
`;
